app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt

```diff
@@ -280,6 +282,15 @@ class TabsTrayFragment : AppCompatDialogFragment() {
         )
     }
 
+    override fun onConfigurationChanged(newConfig: Configuration) {
+        super.onConfigurationChanged(newConfig)
+
+        if (currentOrientation != newConfig.orientation) {
```

Comment:
Since we're adding a `TrayBehaviorManager` that seems to be the only user of `currentOrientation` we could move that state to be managed within the manager (🥁 😄 ).

I don't have strong feelings about this, just an observation. 🙂

Reply:
> This looks nice!
> 
> I think there are some bugs where in landscape mode, we aren't able to see the full tray list. I thought this wouldn't be an issue with #19891 landed already, but I guess there is something missing here?
> 
My branch doesn't yet contain that commit.
After rebasing this will be the result:

https://user-images.githubusercontent.com/11428869/121648810-934ff100-caa0-11eb-8da4-8ba50ffb6c5d.mp4


> The behaviour seems to be allowing the tray to be in the `STATE_HALF_EXPANDED` as well which is strange because it's the same code as before. 🤔
> 
This existed in the old and new tab tray.
Should be an easy fix for which I've opened https://github.com/mozilla-mobile/fenix/issues/19817.